Sources/Sentry/SentryUIApplication.m

```diff
@@ -56,6 +56,90 @@ - (UIApplication *)sharedApplication
     return result;
 }
 
+- (NSArray<UIViewController *> *)relevantViewControllers
+{
+    NSMutableArray *result = [NSMutableArray array];
+
+    NSArray<UIWindow *> *windows = [self windows];
+    if ([windows count] == 0)
+        return nil;
+
+    for (UIWindow *window in windows) {
+        UIViewController *vc = [self relevantViewControllerFromWindow:window];
+        if (vc != NULL) {
+            [result addObject:vc];
+        }
+    }
+
+    return result;
+}
+
+- (UIViewController *)relevantViewControllerFromWindow:(UIWindow *)window
+{
+    UIViewController *topVC = window.rootViewController;
+
+    while (topVC != NULL) {
+        // If the view controller is presenting another one, usually in a modal form.
+        if (topVC.presentedViewController != NULL) {
+            topVC = topVC.presentedViewController;
+            continue;
+            ;
+        }
+
+        // The top view controller is meant for navigation and not content
+        if ([self isHierarchicViewController:topVC]) {
+            topVC = [self relevantViewControllerFromHierarchy:topVC];
+            continue;
+        }
+
+        UIViewController *relevantChild = NULL;
+        for (UIViewController *childVC in topVC.childViewControllers) {
+            // Sometimes a view controller is used as container for a navigation controller
+            // If the navigation is ocuppaing the whole view controller we will consider this the
+            // case.
+            if ([self isHierarchicViewController:childVC]
+                && CGRectEqualToRect(childVC.view.frame, topVC.view.bounds)) {
+                relevantChild = childVC;
+                break;
+            }
+        }
+
+        if (relevantChild != NULL) {
+            topVC = relevantChild;
+            continue;
+        }
+
+        break;
+    }
+
+    return topVC;
+}
+
+- (BOOL)isHierarchicViewController:(UIViewController *)viewController
```

Comment:
`l`: UIKit docs use the wording container for these UIViewControllers, see https://developer.apple.com/documentation/uikit/view_controllers/managing_content_in_your_app_s_windows#3370390.
```suggestion
- (BOOL)isContainerViewController:(UIViewController *)viewController
```